### PR TITLE
Make CombineInputFormat work for Realtime tables

### PR DESCRIPTION
**Pull request by bvaradar**

### CODE REVIEW

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java [hunk 1]
```diff
@@ -149,6 +160,16 @@
...
         combine.setHoodieFilter(true);
         combine.setRealTime(true);
+        if (job.get(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, "").isEmpty()) {
+          List<String> partitions = new ArrayList<>(part.getPartSpec().keySet());
+          if (!partitions.isEmpty()) {
+            String partitionStr = String.join(",", partitions);
+            LOG.info("Setting Partitions in jobConf - Partition Keys for Path : " + path + " is :" + partitionStr);
+            job.set(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, partitionStr);
+          } else {
+            job.set(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, "");
```
Comment (n3nash) [inline hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:170]: Do we need to call a set for else ?

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java [hunk 2]
```diff
@@ -651,16 +690,23 @@ public String toString() {
      */
     @Override
     public void readFields(DataInput in) throws IOException {
-      inputSplitShim.readFields(in);
-      inputFormatClassName = in.readUTF();
+      inputFormatClassName = Text.readString(in);
+      LOG.info("readFields InputFormatClassName : " + inputFormatClassName);
```
Comment (n3nash) [inline hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:694]: May be remove the LOG from here ?

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java [hunk 3]
```diff
@@ -829,18 +882,50 @@ public void createPool(JobConf conf, PathFilter... filters) {
...
-        if (split.getPaths().length > 0) {
-          inputSplitShims.add(new HadoopShimsSecure.InputSplitShim(job, split.getPaths(), split.getStartOffsets(),
-              split.getLengths(), split.getLocations()));
+      if (isRealTime) {
+        job.set("hudi.hive.realtime", "true");
+        InputSplit[] splits;
+        if (hoodieFilter) {
+          HoodieParquetInputFormat input = new HoodieParquetRealtimeInputFormat();
+          input.setConf(job);
+          splits = input.getSplits(job, numSplits);
```
Comment (n3nash) [inline hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:894]: Move this line  splits = input.getSplits(job, numSplits); out of if -else ?

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java (hunk 1 above)
Comment (bvaradar) [reply hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:170]: Yes, From what I understand, the input paths can be across tables (some partitioned and others not so). Hence, resetting before getSplits

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java (hunk 2 above)
Comment (bvaradar) [reply hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:694]: Done

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java (hunk 3 above)
Comment (bvaradar) [reply hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:894]: Not sure if I get it. the else part calls super.getSplits whereas the if part calls input.getSplits. Let me know if I am missing something.

**Comment by bvaradar** [comment]
@n3nash : Addresses/Replied-to review comments. Please take a look when you get a chance.

Thanks,
Balaji.V

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java (hunk 1 above)
Comment (n3nash) [reply hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:170]: ok

File: hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java (hunk 3 above)
Comment (n3nash) [reply hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java:892]: ahh, I missed it, ok thanks

**Pull request merged**